index.js

```diff
@@ -16,7 +17,16 @@ const getWorkflowDir = () => resolveAlfredPrefs().then(prefs => path.join(prefs,
 const readPkg = workflowDir => pathExists(workflowDir)
 	.then(exists => {
 		if (!exists) {
-			throw new Error(`Workflow directory \`${workflowDir}\` does not exist`);
+			return resolveAlfredPrefs()
+				.then(prefs => pathExists(prefs))
+				.then(exists => {
+					if (!exists) {
+						throw new Error(`\`Alfred.alfredpreferences\` package does not exist`);
+					}
+
+					return makeDir(workflowDir)
+						.then(() => readPkgUp());
```

Comment:
Can you move this `.then` to the outer promise chain?

```js
return resolveAlfredPrefs()
	.then(prefs => pathExists(prefs))
	.then(exists => {
		//...		

		return makeDir(workflowDir);
	})
	.then(() => readPkgUp());
```

Also, we have `exists` two times which if I'm not mistaken, should be catched by the latest XO. Rename it to `workflowDirExists` and `preferencesExists` or something.

Reply:
I’ve went a bit further and moved `readPkgUp()` to the outmost promise chain (so we only have one line calling `readPkgUp()`). Also, I’ve reversed the logic (returning early if the workflow directory already exists) to reduce the nesting level. Let me know what you think.

Reply:
Looks good! Thanks!